client/python/gradio_client/templates/discord_chat.py

```diff
@@ -0,0 +1,205 @@
+import asyncio
+import os
+import threading
+from threading import Event
+from typing import Optional
+
+import discord
+import gradio as gr
+from discord import Permissions
+from discord.ext import commands
+from discord.utils import oauth_url
+
+import gradio_client as grc
+from gradio_client.utils import QueueError
+
+event = Event()
+
+DISCORD_TOKEN = os.getenv("DISCORD_TOKEN")
+
+
+async def wait(job):
+    while not job.done():
+        await asyncio.sleep(0.2)
+
+
+def get_client(session: Optional[str] = None) -> grc.Client:
+    client = grc.Client("<<app-src>>", hf_token=os.getenv("HF_TOKEN"))
+    if session:
+        client.session_hash = session
+    return client
+
+
+def truncate_response(response: str) -> str:
+    ending = "...\nTruncating response to 2000 characters due to discord api limits."
+    if len(response) > 2000:
+        return response[: 2000 - len(ending)] + ending
+    else:
+        return response
+
+
+intents = discord.Intents.default()
+intents.message_content = True
+bot = commands.Bot(command_prefix="!", intents=intents)
+
+
+@bot.event
+async def on_ready():
+    print(f"Logged in as {bot.user} (ID: {bot.user.id})")
+    event.set()
+    print("------")
+
+
+thread_to_client = {}
+thread_to_user = {}
+
+
+@bot.command()
+@commands.is_owner()
+async def sync(ctx) -> None:
+    synced = await bot.tree.sync()
+    await ctx.send(f"Synced commands: {', '.join([s.name for s in synced])}.")
+
+
+@bot.hybrid_command(
+    name="<<command-name>>",
+    description="Enter some text to chat with the bot! Like this: /<<command-name>> Hello, how are you?",
+)
+async def chat(ctx, prompt: str):
+    if ctx.author.id == bot.user.id:
+        return
+    try:
+        message = await ctx.send("Creating thread...")
+
+        # User triggered bot via !<<command-name>>
```

Comment:
Is the bot triggered via ! or via commands?

Reply:
It can be both !chat or /chat and the `prompt` function argument is only populated when it's called via `/chat`

Reply:
Cool, thanks!